docs/api-style-guide.md

```diff
@@ -36,6 +36,18 @@ _Note_: We do aware this is different from the common convention. However, we ar
 
 If the resource supports batch operation, then use a separate `/batch` endpoint under that resource.
 
+# Messages
+
+## Property Name Convension
+
+We use json messages to communicate between backend and frontend following [Google JSON Style Guide](https://google.github.io/styleguide/jsoncstyleguide.xml). Property names must be camel-cased, ascii strings. Variable names in different languages should follow their own language styles, e.g. Go and Vue. However, we must use json annotation for every fields in Go API structs to enforce the same style on the wire and prevent any breaking changes by refactoring because Go will set the json property name based on field name automatically.
```

Comment:
nit: camel-cased

Do we mean camelCase or kebab-case actually?

Reply:
I see. But I think that one is pretty confusing.

It's like writing as "red" while using the "blue" as the text color.